src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -92,16 +111,108 @@ protected override Payload ConstructHeadersPayload(BlockLocator locator, uint256
             return headers;
         }
 
-        protected override void AttachCore()
-        {
-            this.AttachedPeer.MessageReceived.Register(this.OnMessageReceivedAsync, true);
-        }
-
         /// <inheritdoc />
         public override object Clone()
         {
             return new ProvenHeadersConsensusManagerBehavior(
-                this.chain, this.initialBlockDownloadState, this.consensusManager, this.peerBanning, this.loggerFactory, this.network);
+                this.chain,
+                this.initialBlockDownloadState,
+                this.consensusManager,
+                this.peerBanning,
+                this.loggerFactory,
+                this.network,
+                this.chainState);
+        }
+
+        /// <summary>
+        /// Determines whether the specified peer supports Proven Headers and PH has been activated.
+        /// </summary>
+        /// <param name="peer">The peer.</param>
+        /// <returns>
+        ///   <c>true</c> if is peer is PH enabled; otherwise, <c>false</c>.
+        /// </returns>
+        private bool IsPeerPhEnabled(INetworkPeer peer)
```

Comment:
PH

Reply:
In pascal cases, acronyms usually are presented by only having the the first letter capitalized, even if there are special rules for acronyms of length 2 that can be both capitalized. Which standard did we have?
Anyway for code consistency I rename it to extended ProvenHeaders